slicedimage/io.py

```diff
@@ -11,24 +11,44 @@
 from six.moves import urllib
 
 from slicedimage.urlpath import pathjoin, pathsplit
-from .backends import CachingBackend, DiskBackend, HttpBackend
+from .backends import CachingBackend, DiskBackend, HttpBackend, SIZE_LIMIT
 from ._collection import Collection
 from ._formats import ImageFormat
 from ._tile import Tile
 from ._tileset import TileSet
 
 
-def infer_backend(baseurl, allow_caching=True):
+def infer_backend(baseurl, backend_config=None):
```

Comment:
Hmm.... I thought of it as the parameters that are being passed along that will eventually make it to the backend (and mused with the idea of allowing passing the backend class itself in the config). Options that come to mind:
 - just `config`
 - `configuration`
 - `backend_setup`

but I'm happy to follow your lead here since I'm having trouble seeing beyond this being the `backend` dictionary out of the `config` object.

Reply:
They are definitely multi-level; I'll clarify in the docs and update to `caching` rather than `cache`.